test/include/seqan3/test/performance/sequence_generator.hpp

```diff
@@ -26,34 +27,84 @@
 namespace seqan3::test
 {
 
-template <typename alphabet_t>
-auto generate_sequence(size_t const len = 500,
-                       size_t const variance = 0,
-                       size_t const seed = 0)
+template <typename sequence_t>
+//!\cond
+    requires (std::ranges::output_range<sequence_t, std::ranges::range_value_t<sequence_t>>)
+//!\endcond
+class random_sequence_generator
 {
-    std::mt19937 gen(seed);
-    size_t max_val{};
-
-    if constexpr (std::unsigned_integral<alphabet_t>)
-        max_val = std::numeric_limits<size_t>::max();
-    else
-        max_val = alphabet_size<alphabet_t> - 1ull;
-
-    std::uniform_int_distribution<size_t> dis_alpha(0ull, max_val);
-    std::uniform_int_distribution<size_t> dis_length(len - variance, len + variance);
+public:
+
+    //!\brief Stores the size of the random sequence.
+    size_t size{};
+    //!\brief Stores the variance of the random sequence.
+    size_t size_variance{};
+
+    random_sequence_generator() = default; //!< Defaulted
+    random_sequence_generator(random_sequence_generator const &) = default; //!< Defaulted
+    random_sequence_generator(random_sequence_generator &&) = default; //!< Defaulted
+    random_sequence_generator & operator=(random_sequence_generator const &) = default; //!< Defaulted
+    random_sequence_generator & operator=(random_sequence_generator &&) = default; //!< Defaulted
+    ~random_sequence_generator() = default; //!< Defaulted
+
+    /*!\brief Initialises a random sequence generator which generates sequences with a given mean size.
+     * \param[in] size          The size of the random sequence.
+     * \param[in] size_variance The variance of the random sequence size.
+     */
+    explicit random_sequence_generator(size_t size, size_t size_variance = 0) :
+        size{size},
+        size_variance{size_variance}
+    {}
+
+    /*!\name Element access
+     * \{
+     */
+
+    /*!\brief Returns a random sequence for a default size and variation.
```

Comment:
```suggestion
    /*!\brief Returns a random sequence for a default size and variation.
```

I'm not completely happy with this sentence, because it is not really a "default size/variation". It is configured by the user and not by us.